src/components/MediaPlayer/VideoJS/VideoJSPlayer.js

```diff
@@ -305,6 +322,96 @@ function VideoJSPlayer({
           console.error('An unknown error occurred.');
           break;
       }
+
+      // Stop retrying if the player is currently in a fallback attempt
+      if (player.isFallingBack) {
+        e.stopPropagation();
+        return;
+      }
+
+      // Determine if this is a multi-source Canvas failure or source quality choice failure
+      /** ASSUMPTION: A Canvas is either multi-sourced OR multi-choice and not both at the same time */
```

Comment:
I assume this assumption is something we'll have to come back to when a use case requires it so thanks for documenting it!